log_downloader.py

```diff
@@ -7,6 +8,22 @@
 import wmi
 
 
+def open_latest_log_in_advantage_scope():
+    try:
+        log_files = [(f, f.stat().st_mtime) for f in LOGS_DIR.glob("*.wpilog")]
+
+        if not log_files:
+            logger.warning("No log files found")
+            return
+
+        latest_log = max(log_files, key=lambda x: x[1])[0]
+
+        logger.info(f"Opening {latest_log.name} with AdvantageScope")
+        subprocess.Popen([ADVANTAGE_SCOPE_PATH, str(latest_log)])
+
+    except Exception as e:
+        logger.error(f"Failed to open log file: {e}")
+
```

Comment:
You know the latest log file. Why find it again here?

Reply:
Where?

Reply:
`download_log_file()`, for example. This is why you should have smaller functions. Have one function find the latest log, another check whether it should be downloaded, then another function download it, and finally, another function open it in the visualizer.

Reply:
I am not finding the latest log file in several places so i think this is fine.

Reply:
I think this is overcomplicated. Just open the file in AdvantageScope after downloading it. Assuming we connect the flash drive after each match, you don't have to check for the latest, as only one will be downloaded.